### PR TITLE
refactor(tail): use variadic tuple type

### PR DESCRIPTION
**Pull request by anuraghazra**
Refactored the older function signature to variadic tuple type introduced in TS 4.0

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -470,11 +470,7 @@ export type XOR<T, U> = T | U extends object ? (Without<T, U> & U) | (Without<U,
 
 /** Functional programming essentials */
 export type Head<T extends AnyArray> = T["length"] extends 0 ? never : T[0];
-export type Tail<T extends AnyArray> = T["length"] extends 0
-  ? never
-  : ((...t: T) => void) extends (first: any, ...rest: infer Rest) => void
-  ? Rest
-  : never;
+export type Tail<T extends AnyArray> = T["length"] extends 0 ? never : T extends [any, ...infer Rest] ? Rest : never;
```
Comment (Beraliv) [inline lib/types.ts:473]: You can even remove first check as it doesn't match next conditional type and will end up being `never` anyway

```suggestion
export type Tail<T extends AnyArray> = T extends [any, ...infer Rest] ? Rest : never;
```

**Review by Beraliv — CHANGES_REQUESTED** [review]
Hey!

Thank you so much for your contribution! Good job and thank you for your changes!

I addressed you with just one comment how it can be even shorter.

Also don't forget to add changelog with `yarn changelog`

File: lib/types.ts (hunk 1 above)
Comment (anuraghazra) [reply lib/types.ts:473]: Aha! Good catch! Thanks.

**Review by Beraliv — APPROVED** [review]
LGTM, thank you for your time and the changes!

**Comment by Beraliv** [comment]
I will merge it as soon as I fix `master` branch CI problems! I will keep you updated!

**Pull request merged**